hexrd/material.py

```diff
@@ -905,32 +909,86 @@ def _set_atomtype(self, v):
         check to make sure number of atoms here is same as
         the atominfo
         """
-        if isinstance(v, list):
-            if len(v) != self.natoms:
-                raise ValueError("incorrect number of atoms")
-        elif isinstance(v, numpy.ndarray):
-            if v.ndim != 1:
-                if v.shape[0] != self.natoms:
-                    raise ValueError("incorrect number of atoms")
-
-        v = numpy.array(v)
-        if self._atomtype.shape == v.shape:
-            if not numpy.allclose(self._atomtype, v):
-                self._atomtype = numpy.array(v)
-                self._newUnitcell()
-                self.update_structure_factor()
-
-            else:
-                return
-        else:
-            self._atomtype = numpy.array(v)
-            self._newUnitcell()
-            self.update_structure_factor()
+        # if isinstance(v, list):
+        #     if len(v) != self.natoms:
+        #         raise ValueError("incorrect number of atoms")
+        # elif isinstance(v, numpy.ndarray):
+        #     if v.ndim != 1:
+        #         if v.shape[0] != self.natoms:
+        #             raise ValueError("incorrect number of atoms")
+
+        # v = numpy.array(v)
+        # if self._atomtype.shape == v.shape:
+        #     if not numpy.allclose(self._atomtype, v):
+        #         self._atomtype = numpy.array(v)
+        #         s#elf._newUnitcell()
+        #         #self.update_structure_factor()
+
+        #     else:
+        #         return
+        # else:
+        #     self._atomtype = numpy.array(v)
+            #self._newUnitcell()
+            #self.update_structure_factor()
+        self._atomtype = v
 
     atomtype = property(
         _get_atomtype, _set_atomtype, None,
         "Information about atomic types")
 
+    def _set_atomdata(self, atomdata):
```

Comment:
I think this would feel a little more natural if the `atomdata` argument were split into three separate arguments (atomtype, atominfo, and U), rather than passing a list of all three.

Reply:
Ok this was part of the latest push.